discgen/utils.py

```diff
@@ -117,6 +124,79 @@ def create_svhn_streams(training_batch_size, monitoring_batch_size):
                           monitoring_batch_size)
 
 
+class Colorize(AgnosticSourcewiseTransformer):
+    def __init__(self, data_stream, **kwargs):
+        super(Colorize, self).__init__(
+             data_stream=data_stream,
+             produces_examples=data_stream.produces_examples,
+             **kwargs)
+
+    def transform_any_source(self, source, _):
+        iters, dim, height, width = source.shape
```

Comment:
Not as long as `source.ndim == 4`. The problem is if `source.ndim == 3`, it'll fail to unpack the 4 values you're asking.